test/test.pl

```diff
@@ -2329,3 +2331,14 @@ sub test_reheader
              err=>'reheader/3_view1.sam.expected.err',
              cmd=>"$$opts{bin}/samtools reheader --in-place $$opts{path}/reheader/hdr.sam $fn.tmp.cram && $$opts{bin}/samtools view -h $fn.tmp.cram | perl -pe 's/\tVN:.*//'");
 }
+
+sub test_addrprg
+{
+    my ($opts,%args) = @_;
+
+    test_cmd($opts,out=>'addrprg/1_fixup.sam.expected', err=>'addrprg/1_fixup.sam.expected.err', cmd=>"$$opts{bin}/samtools addreplacerg -O sam -m overwrite_all $$opts{path}/addrprg/1_fixup.sam -", expect_fail=>0);
+    test_cmd($opts,out=>'addrprg/2_fixup_orphan.sam.expected', err=>'addrprg/2_fixup_orphan.sam.expected.err', cmd=>"$$opts{bin}/samtools addreplacerg -O sam -m orphan_only $$opts{path}/addrprg/2_fixup_orphan.sam -", expect_fail=>0);
+    test_cmd($opts,out=>'addrprg/1_fixup.sam.expected', err=>'addrprg/3_fixup.sam.expected.err', cmd=>"$$opts{bin}/samtools addreplacerg -O sam -R '1#7' $$opts{path}/addrprg/1_fixup.sam -", expect_fail=>1);
+    test_cmd($opts,out=>'addrprg/4_fixup_norg.sam.expected', err=>'addrprg/4_fixup_norg.sam.expected.err', cmd=>"$$opts{bin}/samtools addreplacerg -O sam -r '\@RG\\tID:1#8\\tCN:SC' $$opts{path}/addrprg/4_fixup_norg.sam -", expect_fail=>0);
+    test_cmd($opts,out=>'addrprg/1_fixup.sam.expected', err=>'addrprg/1_fixup.sam.expected.err', cmd=>"$$opts{bin}/samtools addreplacerg -O sam -m overwrite_all -R '1#8' $$opts{path}/addrprg/1_fixup.sam -", expect_fail=>0);
+}
```

Comment:
`expect_fail` is our way of temporarily saying "this test is broken".  So the ways this should appear in `test_cmd()` are (1) not at all, or (2) as `expect_fail=>1` to signal a broken test.  So please don't write `expect_fail=>0` everywhere as it's just noise.